docs/src/README.md

```diff
@@ -0,0 +1,123 @@
+
+`getFunnelForToken(address _tokenAddress)` - Returns the funnel contract address for a given token address
+
+`isFunnel(address _funnelAddress)` - Returns true if the funnel contract is a funnel deployed by the factory.
+
+### Funnel.sol - Funnel contract for ERC20 tokens
+
+`baseToken()` - Returns the address of the underlying token
+
+# Usage
+
+## Testing
+
+Our tests consist of both Foundry tests and hardhat tests. 
+
+`forge test` - Runs the Foundry tests. Forge gas snapshots will be captured under `.forge-snapshots`
+
+`npx hardhat test` - Runs hardhat tests
+
+`npm test` - Runs all tests
+
+## Deployment
+
+Make copy of `.env.template` to `.env` and fill in the values.
+
+Running local fork
+`anvil`
+
+Deploy to local fork
+
+`forge script script/FunnelFactoryDeployer.sol:FunnelFactoryDeployer --fork-url http://localhost:8545 --broadcast`
+
+Deploy factory to goerli
+
+`forge script script/FunnelFactoryDeployer.sol:FunnelFactoryDeployer --rpc-url $GOERLI_RPC_URL --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY`
+
+Note: Deployment addresses are dependent on FunnelFactory bytecode and salt used. Multiple deployments will fail.
+
+## Verification
+
+Private verification
+`npx hardhat run script/verifyTenderly.ts --network goerli`
+
+## Misc
+
+Run prettier to format the code
+
+```sh
+npx prettier --write 'src/**/*.sol'
+```
+
+### Gas Report
+
+Forge snapshot automatically captures selected function calls on `.forge-snapshots`. To generate a more comprehensive report, run:
+
+```sh
+forge test --gas-report >> .forge-snapshots/report.txt   
+forge snapshot # needed for gas checks
+```
+
+### Test Coverage
+
+```sh
+forge coverage --report lcov   
+genhtml lcov.info -o report --branch-coverage 
+```
+
+### Documentation
+
+Documentation is generated automatically from the source files and will be available at `docs/`. To regenerate the documentation, run:
+
+```sh
+forge doc --serve --port 4000   
+```
+
+# Deployments
+
+Script for batch deployment across testnet or mainnet
+`./script/deployAll.sh`
+
+| Network   | Contract      | Address                                    | Version     |
+| --------- | ------------- | ------------------------------------------ | ----------- |
+| Goerli    | Funnel (impl) | 0xb2acc9a328039e2918b2d85420251fb831ce47e0 | 0.2.0-alpha |
+| Goerli    | FunnelFactory | 0x92324acad3946f0600d7e0e2641ef10d1655d89d | 0.2.0-alpha |
+| Polygon   | Funnel (impl) | 0xb2acc9a328039e2918b2d85420251fb831ce47e0 | 0.2.0-alpha |
+| Polygon   | FunnelFactory | 0x92324acad3946f0600d7e0e2641ef10d1655d89d | 0.2.0-alpha |
+| Avalanche | Funnel (impl) | 0xb2acc9a328039e2918b2d85420251fb831ce47e0 | 0.2.0-alpha |
+| Avalanche | FunnelFactory | 0x92324acad3946f0600d7e0e2641ef10d1655d89d | 0.2.0-alpha |
+| Arbitrum  | Funnel (impl) | 0xb2acc9a328039e2918b2d85420251fb831ce47e0 | 0.2.0-alpha |
+| Arbitrum  | FunnelFactory | 0x92324acad3946f0600d7e0e2641ef10d1655d89d | 0.2.0-alpha |
+
```

Comment:
Would prefer this section to be in the root README since it's the key info devs need after we deploy.